### PR TITLE
feat: added watch video & history pages

### PR DESCRIPTION
**Pull request by kedark152**
### feat: added watch video & history pages

Features:-

- Added Single Video page which opens with unique link.
- added like, watch later, add to playlist buttons on Single video page.
- added history page with  remove a video from history and clear all history features.

**Code Update after review:- (14/05/22)**
- created two new custom hooks to avoid code repeatability: useSingleVideoBtns & useVideoOnClickHandler.
- useSingleVideoBtns  returns the state of like, watch later buttons after checking like list & watch later list in videoActionState
- useVideoOnClickHandler returns setState function is used to update the state, & run the required function on Click.
- In singlevideo.js logic/validations are reduced to separate file using custom hook.
- In video options model box (videoOptions.js) component, used new custom hooks.
- Added video to history on SingleVideo.js after page gets loaded, instead of adding on click of video card in home page.
-  stored constant variable names to constants.js as these values as constants for avoiding any typo errors

Preview Link: https://deploy-preview-6--agroplay.netlify.app/

### CODE REVIEW

**Comment by bajajcodes** [comment]
## Suggestions

- In singlevideo.js logic/validations can be reduced to separate file using custom hook
- Write something like checkauth.js fof auto redirection for login pges(smiliar to requires auth, but for auth pages)

## Liked

- Use of token in state.
- Descriptiive meanigful naming.
- Use of loader for doing api calls.
- Use of single reducer app, will think to implement to in next app.

*Explorer page is not working*

File: src/pages/SingleVideo.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        "/login",
+        { state: { from: { pathname: pathname } } },
+        { replace: true }
+      );
+    }
+  };
+
+  const onActionClickHandler = (id) => {
+    switch (id) {
+      case "like-video":
```
Comment (ArjunGTX) [inline src/pages/SingleVideo.js]: You can store these values as constants so that there wont be any typo errors

File: src/pages/SingleVideo.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+                  frameBorder="0"
+                  allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+                  allowFullScreen
+                ></iframe>
+              </div>
+              <div className="video-action-buttons">
+                <ul className="flex fw-bold ">
+                  <li
+                    id="like-video"
+                    onClick={(e) => onClickHandler(e.target.id)}
```
Comment (ArjunGTX) [inline src/pages/SingleVideo.js]: you can directly pass the event object and avoid the arrow function here

**Review by ArjunGTX — APPROVED** [review]

**Comment by kedark152** [comment]
Thanks Shubham & Arjun for detailed PR Review.
Have made some suggested changes, details updated in PR description.
Merging the PR.

**Pull request merged**